azurerm/resource_arm_app_service_certificate.go

```diff
@@ -0,0 +1,250 @@
+				ForceNew:     true,
+				ValidateFunc: validate.Base64String(),
+			},
+
+			"issuer": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"issue_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"expiration_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"password": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Sensitive:    true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			"thumbprint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"key_vault_id": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc:     azure.ValidateResourceID,
+				ConflictsWith:    []string{"pfx_blob", "password"},
+			},
+
+			"key_vault_secret_name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ValidateFunc:  validate.NoEmptyStrings,
+				ConflictsWith: []string{"pfx_blob", "password"},
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmAppServiceCertificateCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).web.CertificatesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for App Service Certificate creation.")
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	password := d.Get("password").(string)
+	t := d.Get("tags").(map[string]interface{})
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Certificate %q (Resource Group %q): %s", name, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_app_service_certificate", *existing.ID)
+		}
+	}
+
+	certificate := web.Certificate{
+		CertificateProperties: &web.CertificateProperties{
+			Password: utils.String(password),
+		},
+		Location: utils.String(location),
+		Tags:     tags.Expand(t),
+	}
+
+	if v, ok := d.GetOk("pfx_blob"); ok {
+		pfxBlob, err := base64.StdEncoding.DecodeString(v.(string))
+		if err != nil {
+			return fmt.Errorf("Could not decode PFX blob: %+v", err)
+		}
+		certificate.CertificateProperties.PfxBlob = &pfxBlob
+	}
+
+	if v, ok := d.GetOk("key_vault_id"); ok {
+		certificate.CertificateProperties.KeyVaultID = utils.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("key_vault_secret_name"); ok {
+		certificate.CertificateProperties.KeyVaultSecretName = utils.String(v.(string))
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, name, certificate); err != nil {
+		return err
```

Comment:
can we wrap this error e.g.

```suggestion
		return fmt.Errorf("Error creating/updating App Service Certificate %q (Resource Group %q): %s", name, resourceGroup, err)
```